### PR TITLE
[Reviewer: EM] Rename SIP Connection Pool class

### PR DESCRIPTION
**Pull request by TJW2**
Rename the ConnectionPool class in include/connection_pool.h to SipConnectionPool to avoid a conflict with the template ConnectionPool class in cpp-common introduced in pull request [#546](https://github.com/Metaswitch/cpp-common/pull/546).

### CODE REVIEW

File: modules/cpp-common [hunk 1]
```diff
@@ -1 +1 @@
-Subproject commit e806b651a4ed6e24828849850235a526b6796d03
+Subproject commit 8a2e0b9b4f25b4cf5206e97e1df8785da5708507
```
Comment (TJW2) [inline modules/cpp-common]: Is this change something to be worried about? I'm not sure what it's doing.

File: include/sipconnectionpool.h [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /**
- * @file connection_pool.h Class for maintaining a SIP connection pool.
+ * @file sipconnectionpool.h Class for maintaining a SIP connection pool.
```
Comment (eleanor-merry) [inline include/sipconnectionpool.h]: Can this be sip_connection_pool rather than sipconnectionpool?

File: src/sipconnectionpool.cpp [hunk 3]
```diff
@@ -1,5 +1,5 @@
 /**
- * @file connection_pool.cpp
+ * @file sipconnectionpool.cpp
```
Comment (eleanor-merry) [inline src/sipconnectionpool.cpp]: Likewise, can this be sip_connection_pool as well?

File: docs/Bono Sprout Architecture.md [hunk 4]
```diff
@@ -120,7 +120,7 @@ When the stateful proxy module is running as a Bono node, it performs various ed
 
 When running as a Bono node all routing of requests is performed based on route headers and request URI.  If a request cannot be routed based on route headers or the request URI then Bono will always route it to a Sprout node - it does not access the registration store directly itself.  This means that a Bono node will never fork a SIP request.
 
-Each Bono node uses the ConnectionPool class to manage a pool of TCP connections to the Sprout nodes for this purpose.  The ConnectionPool class periodically recycles these connections to keep the load evenly spread across the Sprout nodes.
+Each Bono node uses the SipConnectionPool class to manage a pool of TCP connections to the Sprout nodes for this purpose.  The SipConnectionPool class periodically recycles these connections to keep the load evenly spread across the Sprout nodes.
```
Comment (eleanor-merry) [inline docs/Bono Sprout Architecture.md]: I think I'd rather SIPConnectionPool rather than SipConnectionPool, but I don't really care that much

**Comment by eleanor-merry** [comment]
LGTM

File: docs/Bono Sprout Architecture.md (hunk 4 above)
Comment (TJW2) [inline docs/Bono Sprout Architecture.md]: Here I'm using the style used by HttpConnection - is there a convention for this?

File: include/sipconnectionpool.h (hunk 2 above)
Comment (TJW2) [inline include/sipconnectionpool.h]: sipconnectionpool is consistent with sipresolver, although the rest of the files in sprout/src are named with a mixture of both styles. Should I make this change anyway?

Comment (eleanor-merry) [inline include/sipconnectionpool.h]: I think we've settled on using _ now (and new files follow this pattern), so can you make the change?

**Pull request merged**